### PR TITLE
Chore: Refactor data-fetching and simplify ApplicationInfosPage

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

- Refactor data-fetching to use react-query
- Simplify utils (one more lodash util down 🔫  )

### Why is it needed?

Housekeeping

### CODE REVIEW

File: packages/core/admin/admin/src/pages/SettingsPage/index.js [hunk 1]
```diff
@@ -79,5 +68,5 @@ function SettingsPage() {
   );
 }
 
-export default memo(SettingsPage);
-export { SettingsPage };
+// Required for React.lazy
+export default SettingsPage;
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/pages/SettingsPage/index.js:72]: is it?

Comment (gu-stav) [reply packages/core/admin/admin/src/pages/SettingsPage/index.js]: No, it is fake news and I have fact-checked + rebased.

File: packages/core/admin/admin/src/pages/SettingsPage/index.js [hunk 2]
```diff
@@ -43,13 +32,17 @@ function SettingsPage() {
...
       routes.map(({ to, Component, exact }) => createRoute(Component, to, exact))
     );
   }, [routes]);
 
-  const pluginsRoutes = createSectionsRoutes(settings);
+  const pluginsRoutes = Object.values(settings).flatMap((section) => {
+    const { links } = section;
+
+    return links.map((link) => createRoute(link.Component, link.to, link.exact || false));
+  });
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/pages/SettingsPage/index.js:45]: I would inline this to the return part and not use the `createRoute` utility, `createRoute` is a very lazy abstraction imo

Comment (joshuaellis) [reply packages/core/admin/admin/src/pages/SettingsPage/index.js:45]: sure 🏁

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**